lib/oban/queue/executor.ex

```diff
@@ -36,9 +36,35 @@ defmodule Oban.Queue.Executor do
   end
 
   @doc false
-  def safe_call(%Job{args: args, worker: worker} = job) do
+  def safe_call(%Job{} = job) do
+    with {:ok, worker} <- worker_module(job) do
+      case worker.timeout(job) do
+        :infinity ->
+          perform_inline(worker, job)
```

Comment:
If the timeout is `:infinity`, I preserve running the work in the current process which eliminates pointless copying to the task process.